### PR TITLE
Add `positive_score_impact` support for `rank_features`

### PR DESCRIPTION
**Pull request by Hronom**
### Description
Added support for `positive_score_impact` parameter for `rank_features`
 
### Issues Resolved
[[List any issues this PR will resolve]](https://github.com/opensearch-project/OpenSearch/issues/2724)
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Hronom** [comment]
start gradle check

**Comment by Hronom** [comment]
Admins please restart build... since failed test completed ok locally and changes also works good here https://github.com/opensearch-project/OpenSearch/pull/2725

**Review by nknize — CHANGES_REQUESTED** [review]
We will not be releasing 1.4 and 2.0 is feature frozen so the earliest we can merge this will be for 2.1. Let's work on getting it in main first and then we'll autobackport to 2.x.

File: modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java [hunk 1]
```diff
@@ -84,16 +98,23 @@ public RankFeaturesFieldMapper build(BuilderContext context) {
 
     public static final class RankFeaturesFieldType extends MappedFieldType {
 
-        public RankFeaturesFieldType(String name, Map<String, String> meta) {
+        private final boolean positiveScoreImpact;
```
Comment (reta) [inline modules/mapper-extras/src/main/java/org/opensearch/index/mapper/RankFeaturesFieldMapper.java:101]: It seems like `positiveScoreImpact` is not needed for `RankFeaturesFieldType`, the `positiveScoreImpact()` is never used

**Comment by nknize** [comment]
Please fold the above PR feedback into #2725. We will not be merging this into 1.x since there will be no 1.4 release.

**Comment by dblock** [comment]
I'll close this and leave https://github.com/opensearch-project/OpenSearch/pull/2725. Once in main we can talk about how far we want to backport it.

**Pull request closed without merging**